builder/context.go

```diff
@@ -35,15 +35,17 @@ func (b *Builder) getDockerRunArgs(
 	step *graph.Step,
 	envs []string,
 	entrypoint string,
-	cmd string) []string {
-
+	cmd string,
+	containerName string) []string {
 	var args []string
 	var sb strings.Builder
+	addWorkdir := true
+
 	// Run user commands from a shell instance in order to mirror the shell's field splitting algorithms,
 	// so we don't have to write our own argv parser for exec.Command.
 	if runtime.GOOS == windowsOS {
 		args = []string{"powershell.exe", "-Command"}
-		if step.Isolation == "" && !step.IsBuildStep() {
+		if step != nil && step.Isolation == "" && !step.IsBuildStep() {
```

Comment:
>step != nil [](start = 5, length = 11)

what is the new case that can pass null step to the function?

Reply:
I assume the function is only called to create run args for a step. If the step is null, it make me think if it's better to create a new function to handle the case instead of reusing the existing one. As you see, the function starts to have more branches and it's warning sign for maintainability. 

---
In reply to: [307847029](https://github.com/Azure/acr-builder/pull/469#discussion_r307847029) [](ancestors = 307847029)

Reply:
yep, i have created a new umbrella function to create run arguments for docker. That will also be internally used by creating run args for Step.